src/test/java/com/alkemy/ong/service/impl/ActivityServiceImplTest.java

```diff
@@ -0,0 +1,168 @@
+        activityDto.setImage( "Image" );
+        activityDto.setName( "Name" );
+        return activityDto;
+    }
+
+    private Activity getActivityWithId() {
+
+        Activity activity = getActivity( false );
+        activity.setId( "algo" );
+        return activity;
+    }
+
+    private static Activity getActivity(boolean softDelete) {
+
+        Activity activity = new Activity();
+        activity.setContent( "Not all who wander are lost" );
+        activity.setId( "42" );
+        activity.setImage( "Image" );
+        activity.setName( "Name" );
+        activity.setDeleted( softDelete );
+        activity.setTimestamp( Timestamp.from( LocalDateTime.of( 1, 1, 1, 1, 1, 1 ).toInstant( ZoneOffset.UTC ) ) );
+        return activity;
+    }
+
+    private Activity getUpdatedTestimonial() {
+
+        Activity activityWithId = getActivityWithId();
+        activityWithId.setId( "42" );
+        activityWithId.setName( "updated" );
+        activityWithId.setImage( "updated" );
+        activityWithId.setContent( "updated" );
+        activityWithId.setDeleted( false );
+        return activityWithId;
+    }
+
+    @Test
+    void UpdateActivityAndReturnEntityUpdated() {
+
+        Activity activity = new Activity();
+        activity.setContent( "Not all who wander are lost" );
+        activity.setId( "42" );
+        activity.setImage( "Image" );
+        activity.setName( "Name" );
+        activity.setDeleted( false );
+        activity.setTimestamp( Timestamp.from( LocalDateTime.of( 1, 1, 1, 1, 1, 1 ).toInstant( ZoneOffset.UTC ) ) );
+
+        when( activityRepository.findById( anyString() ) ).thenReturn( Optional.of( activity ) );
+        ActivityDto activityDto = getActivityDto();
+        activityDto.setContent( "updated" );
+        activityDto.setImage( "updated" );
+        activityDto.setName( "updated" );
+        Activity value = new Activity();
+        value.setName( activityDto.getName() );
+        value.setContent( activityDto.getContent() );
+        value.setImage( activityDto.getImage() );
+        when( activityRepository.save( any() ) ).thenReturn( value );
+
+        Activity updateActivity = activityServiceImpl.crearActivity( activity);
+
+        assertThat( updateActivity.getName() ).isEqualTo( activityDto.getName() ).isNotEqualTo( activity.getName() );
+
+        assertThat( updateActivity.getImage() ).isEqualTo( activityDto.getImage() ).isNotEqualTo( activity.getImage() );
+
+        assertThat( updateActivity.getContent() ).isEqualTo( activityDto.getContent() ).isNotEqualTo( activity.getContent() );
+
+    }
+
+    @Test
+    void CreateActivityAndReturnEntitySaved() {
+
+        Activity activity = getActivity( true );
+        Activity activityWithId = getActivityWithId();
+        ArgumentCaptor<Activity> activity2 = ArgumentCaptor.forClass( Activity.class );
+        when( activityRepository.save( activity2.capture() ) ).thenReturn( activityWithId );
```

Comment:
este `activity2.capture() ` deberia ser remplazado por `any()` y mover ese `activity2.capture()` a   `verify( activityRepository ).save( any() );`
y te explico porque; segun documentacion esta aconsejado utilizar el capture en los verify. [Documentacion](https://site.mockito.org/javadoc/current/org/mockito/ArgumentCaptor.html)

Y obviamente tenes que mover los assert que dependan de esta variable.
```suggestion
         when( activityRepository.save( any()) ).thenReturn( activityWithId );
```